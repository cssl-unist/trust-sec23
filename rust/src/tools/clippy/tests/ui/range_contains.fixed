// run-rustfix

#[warn(clippy::manual_range_contains)]
#[allow(unused)]
#[allow(clippy::no_effect)]
#[allow(clippy::short_circuit_statement)]
#[allow(clippy::unnecessary_operation)]
fn main() {
    let x = 9_u32;

    // order shouldn't matter
    (8..12).contains(&x);
    (21..42).contains(&x);
    (1..100).contains(&x);

    // also with inclusive ranges
    (9..=99).contains(&x);
    (1..=33).contains(&x);
    (1..=999).contains(&x);

    // and the outside
    !(8..12).contains(&x);
    !(21..42).contains(&x);
    !(1..100).contains(&x);

    // also with the outside of inclusive ranges
    !(9..=99).contains(&x);
    !(1..=33).contains(&x);
    !(1..=999).contains(&x);

    // not a range.contains
    x > 8 && x < 12; // lower bound not inclusive
    x < 8 && x <= 12; // same direction
    x >= 12 && 12 >= x; // same bounds
    x < 8 && x > 12; // wrong direction

    x <= 8 || x >= 12;
    x >= 8 || x >= 12;
    x < 12 || 12 < x;
    x >= 8 || x <= 12;
}
