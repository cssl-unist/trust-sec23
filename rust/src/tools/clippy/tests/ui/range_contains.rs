// run-rustfix

#[warn(clippy::manual_range_contains)]
#[allow(unused)]
#[allow(clippy::no_effect)]
#[allow(clippy::short_circuit_statement)]
#[allow(clippy::unnecessary_operation)]
fn main() {
    let x = 9_u32;

    // order shouldn't matter
    x >= 8 && x < 12;
    x < 42 && x >= 21;
    100 > x && 1 <= x;

    // also with inclusive ranges
    x >= 9 && x <= 99;
    x <= 33 && x >= 1;
    999 >= x && 1 <= x;

    // and the outside
    x < 8 || x >= 12;
    x >= 42 || x < 21;
    100 <= x || 1 > x;

    // also with the outside of inclusive ranges
    x < 9 || x > 99;
    x > 33 || x < 1;
    999 < x || 1 > x;

    // not a range.contains
    x > 8 && x < 12; // lower bound not inclusive
    x < 8 && x <= 12; // same direction
    x >= 12 && 12 >= x; // same bounds
    x < 8 && x > 12; // wrong direction

    x <= 8 || x >= 12;
    x >= 8 || x >= 12;
    x < 12 || 12 < x;
    x >= 8 || x <= 12;
}
