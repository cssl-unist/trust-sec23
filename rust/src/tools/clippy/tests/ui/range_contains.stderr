error: manual `Range::contains` implementation
  --> $DIR/range_contains.rs:12:5
   |
LL |     x >= 8 && x < 12;
   |     ^^^^^^^^^^^^^^^^ help: use: `(8..12).contains(&x)`
   |
   = note: `-D clippy::manual-range-contains` implied by `-D warnings`

error: manual `Range::contains` implementation
  --> $DIR/range_contains.rs:13:5
   |
LL |     x < 42 && x >= 21;
   |     ^^^^^^^^^^^^^^^^^ help: use: `(21..42).contains(&x)`

error: manual `Range::contains` implementation
  --> $DIR/range_contains.rs:14:5
   |
LL |     100 > x && 1 <= x;
   |     ^^^^^^^^^^^^^^^^^ help: use: `(1..100).contains(&x)`

error: manual `RangeInclusive::contains` implementation
  --> $DIR/range_contains.rs:17:5
   |
LL |     x >= 9 && x <= 99;
   |     ^^^^^^^^^^^^^^^^^ help: use: `(9..=99).contains(&x)`

error: manual `RangeInclusive::contains` implementation
  --> $DIR/range_contains.rs:18:5
   |
LL |     x <= 33 && x >= 1;
   |     ^^^^^^^^^^^^^^^^^ help: use: `(1..=33).contains(&x)`

error: manual `RangeInclusive::contains` implementation
  --> $DIR/range_contains.rs:19:5
   |
LL |     999 >= x && 1 <= x;
   |     ^^^^^^^^^^^^^^^^^^ help: use: `(1..=999).contains(&x)`

error: manual `!Range::contains` implementation
  --> $DIR/range_contains.rs:22:5
   |
LL |     x < 8 || x >= 12;
   |     ^^^^^^^^^^^^^^^^ help: use: `!(8..12).contains(&x)`

error: manual `!Range::contains` implementation
  --> $DIR/range_contains.rs:23:5
   |
LL |     x >= 42 || x < 21;
   |     ^^^^^^^^^^^^^^^^^ help: use: `!(21..42).contains(&x)`

error: manual `!Range::contains` implementation
  --> $DIR/range_contains.rs:24:5
   |
LL |     100 <= x || 1 > x;
   |     ^^^^^^^^^^^^^^^^^ help: use: `!(1..100).contains(&x)`

error: manual `!RangeInclusive::contains` implementation
  --> $DIR/range_contains.rs:27:5
   |
LL |     x < 9 || x > 99;
   |     ^^^^^^^^^^^^^^^ help: use: `!(9..=99).contains(&x)`

error: manual `!RangeInclusive::contains` implementation
  --> $DIR/range_contains.rs:28:5
   |
LL |     x > 33 || x < 1;
   |     ^^^^^^^^^^^^^^^ help: use: `!(1..=33).contains(&x)`

error: manual `!RangeInclusive::contains` implementation
  --> $DIR/range_contains.rs:29:5
   |
LL |     999 < x || 1 > x;
   |     ^^^^^^^^^^^^^^^^ help: use: `!(1..=999).contains(&x)`

error: aborting due to 12 previous errors

